lib/uuid4.rb

```diff
@@ -95,17 +95,22 @@ def new(value = nil)
       end
     end
 
+    def valid?(value)
+      new(value)
+      true
+    rescue TypeError
```

Comment:
I would expect that only the UUID4 specific exception should be rescued. Therefore I highly recommend introducing a specific `TypeError`  subclass exception and throw/rescue that exception.